instrumentation/apache-camel-2.20/src/test/groovy/test/TwoServicesCamelSpringBootBasedTest.groovy

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package test
+
+import static io.opentelemetry.trace.Span.Kind.CLIENT
+import static io.opentelemetry.trace.Span.Kind.INTERNAL
+import static io.opentelemetry.trace.Span.Kind.SERVER
+
+import com.google.common.collect.ImmutableMap
+import io.opentelemetry.auto.test.AgentTestRunner
+import io.opentelemetry.auto.test.utils.PortUtils
+import io.opentelemetry.trace.attributes.SemanticAttributes
+import org.apache.camel.CamelContext
+import org.apache.camel.ProducerTemplate
+import org.apache.camel.builder.RouteBuilder
+import org.apache.camel.impl.DefaultCamelContext
+import org.springframework.boot.SpringApplication
+import org.springframework.context.ConfigurableApplicationContext
+import spock.lang.Shared
+
+class TwoServicesCamelSpringBootBasedTest extends AgentTestRunner {
+
+  @Shared
+  int portOne = PortUtils.randomOpenPort()
+  @Shared
+  int portTwo = PortUtils.randomOpenPort()
+  @Shared
+  ConfigurableApplicationContext server
+  @Shared
+  CamelContext clientContext
+
+  def setupSpec() {
+    createServer()
+  }
+
+  def createServer() {
+    def app = new SpringApplication(TwoServicesConfig)
+    app.setDefaultProperties(ImmutableMap.of("service.one.port", portOne, "service.two.port", portTwo))
+    server = app.run()
+  }
+
+  def createAndStartClient() {
+    clientContext = new DefaultCamelContext()
+    clientContext.addRoutes(new RouteBuilder() {
+      void configure() {
+        from("direct:input")
+          .log("SENT Client request")
+          .to("http://localhost:$portOne/serviceOne")
+          .log("RECEIVED Client response")
+      }
+    })
+    clientContext.start()
+  }
+
+  def cleanupSpec() {
+    if (server != null) {
+      server.close()
+      server = null
+    }
+  }
+
+  def "two camel service spans"() {
+    setup:
+    createAndStartClient()
+    ProducerTemplate template = clientContext.createProducerTemplate()
+
+    when:
+    template.sendBody("direct:input", "Example request")
+
+    then:
+    assertTraces(1) {
+      trace(0, 9) {
+        it.span(0) {
+          name "input"
+          kind CLIENT
```

Comment:
This is wrong too, there's no `SERVER` child for that span. `direct` camel component isn't really a server by the way, so maybe this trace should start with an `INTERNAL` span? It's not really a real-life scenario anyway, since you can only call `direct:*` from inside.

Reply:
That's a really good question. 
IMO setting "direct" as CLIENT is OK, as "direct" is a way to call a route (with remote services in line). We may model this as a Camel equivalent of a HttpClient. However as you pointed out - this does not go in line with SemConv recommendation. There may be some client side processing - producing INTERNAL spans before equivalent SERVER span is created. Heck, the route can be entirely local (without any SERVER spans). Same would apply to eg "timer" spans. 
Anyone else would like to comment on that matter? @iNikem ? @trask ? :)

Reply:
Decided to reimplement internal components as "INTERNAL" spans. Data produced looks much better now.